### PR TITLE
Create add-rat-in-a-maze.md

### LINKED ISSUE #1333: Add rat in a maze
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/ajay-dhangar/algo/blob/main/CONTRIBUTING.md).

### Description

Problem Description: A precise explanation of the problem's goal, input constraints, and expected output.
Algorithm Explanation: A step-by-step breakdown of the backtracking approach, including why certain paths are chosen or discarded.
Example Walkthrough: An example maze with visual guidance on how the algorithm explores paths, makes decisions, and backtracks.
Code Documentation: Inline comments in the code to clarify each function and significant code sections.
Edge Case Handling: Information on handling edge cases like blocked mazes, already-at-destination scenarios, and empty mazes.

### Self-service

- [X] I'd be willing to address this documentation request myself.

### PR DESCRIPTION
**Pull request by ananyag309**
### Description
Added rat in a maze algorithm
Fixes #1333 
### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/DSA-Problem-Solution/add-rat-in-a-maze.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
+
+
+# Rat in a Maze - Backtracking Solution
+
```
Comment (ajay-dhangar) [inline docs/DSA-Problem-Solution/add-rat-in-a-maze.md]: ⁉️

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
careful read our contribution guidelines

File: docs/DSA-Problem-Solution/add-rat-in-a-maze.md (hunk 1 above)
Comment (ananyag309) [reply docs/DSA-Problem-Solution/add-rat-in-a-maze.md:4]: what is the issue?

**Comment by ananyag309** [comment]
@ajay-dhangar i am extremely sorry for all this, i will be raising the correct pr

**Pull request closed without merging**